nncf/common/pruning/export_helpers.py

```diff
@@ -10,43 +10,226 @@
         """
-        Propagates the pruning mask through a node using pruning masks of all inputs and the current node (if any).
+        Return whether all input sources of node is convolutions or not.
 
-        :param node: The graph node to propagate mask through it
-        :param graph: The model graph to prune
+        :param node: Node to determine it's sources
+        :param graph: NNCF graph to work with
+        :return: True if all input sources of node is convolutions
         """
-        raise NotImplementedError
+
+        for input_node in graph.get_previous_nodes(node):
+            # If input has mask ->  it went from convolution (source of this node is a convolution)
+            if input_node.data.get('output_mask', None) is not None:
+                continue
+
+            source_nodes = get_sources_of_node(input_node, graph, cls.ConvolutionOp.get_all_op_aliases() +
+                                               cls.StopMaskForwardOp.get_all_op_aliases() +
+                                               cls.InputOp.get_all_op_aliases())
+            sources_types = [node.node_type for node in source_nodes] + [input_node.node_type]
+            if any(t in sources_types for t in cls.StopMaskForwardOp.get_all_op_aliases()):
+                return False
+        return True
+
+    @classmethod
+    def _get_unit_mask(cls, dim, device):
+        return np.ones(dim)
+
+    @classmethod
+    def _get_masks_device(cls, input_masks):
+        return None
 
     @classmethod
-    def get_all_op_aliases(cls):
+    def _concat_masks(cls, filled_input_masks):
+        return np.concatenate(filled_input_masks, 0)
+
+    @classmethod
+    def generate_output_mask(cls, node: NNCFNode, graph: NNCFGraph) -> Union[List[np.array], None]:
         """
-        :return: list of all aliases of types in metatype
+        Generate output mask from input masks with all None replaced by identity masks.
+        If all input masks is None return None.
+
+        :param node: Node to determine it's sources.
+        :param graph: NNCF graph to work with.
+        :return: Filled input masks.
         """
-        op_types = []
-        for subtype in cls.subtypes:
-            op_types.extend(subtype.get_all_aliases())
-        op_types = list(set(op_types)) + cls.additional_types
-        return op_types
+        input_edges = graph.get_input_edges(node)
+        previous_nodes = [edge.from_node for edge in input_edges]
+        input_masks = [input_node.data['output_mask'] for input_node in previous_nodes]
+
+        if all(mask is None for mask in input_masks):
+            return None
+
+        filled_input_masks = []
+        for i, mask in enumerate(input_masks):
+            if mask is None:
+                concat_axis = node.layer_attributes.axis
+                concat_dim = input_edges[i].tensor_shape[concat_axis]
+                device = cls._get_masks_device(input_masks)
+                mask = cls._get_unit_mask(concat_dim, device)
```

Comment:
After concat layer attributes implementation I realized one can just use shape of the first non empty input masks because all input masks **should** be the same and thus concat layer attributes is redundant. But current approach is a bit more straightforward. If you would like I'll remove concat axis attribute and everything will work ok. But maybe concat axis will help in the future

Reply:
I think Concat axis attribute can be helpful.

Reply:
Discussed with @daniil-lyakhov offline that **axis** is sufficient since inputs can have different shapes on concatenated axis, for example: 
Concat([1, 2, 5, 6], [1, 3, 5, 6], [1, 4, 5, 6], axis=1) -> [1, 2+3+4, 5, 6]